### ISSUE TITLE
Allow semicolon omission

### ISSUE THREAD

**Issue by dani-h** [leafgarland/typescript-vim#39]
Currently when not adding a semicolon and entering a newline the next line will be indented one step. If you however add a semicolon to the first statement the next line will not be indented as expected. Is it possible to have a setting for this? I usually omit semicolons and have to delete a few spaces to return to the neutral position.

Ex1:
```
console.log("Hello");
console.log("World"); //Works, no indentation
```

Ex2:
```
console.log("Hello")
  console.log("World")
```

**Comment by isaachess** [comment]
+1 on this! I was about to submit this exact same issue.

On your readme it says this plugin no longer provides any indentation support, and instead relies on either the native js indenting or a plugin like vim-js-indent. This problem is super annoying, and I would really _like_ for the typescript plugin to have no opinion about ts indenting, because I do not have this problem in my javascript files.

**Comment by isaachess** [comment]
@dani-h If you're curious, I got around this issue by installing `vim-js-indent` (https://github.com/jason0x43/vim-js-indent) and then commented-out his entire `/indent/typescript.vim` file. This removed all his indent stuff, and instead lets the `vim-js-indent` plugin handle it, which is nice.

@leafgarland It seems a little weird that you state the plugin does no indent work, but has an entire file dedicated to indentation.

**Comment by dani-h** [comment]
@isaachess Oh, I don't edit much javascript so `autocmd FileType typescript,javascript setlocal expandtab shiftwidth=2 tabstop=2` tends to work. For indenting whole files I use Chiel92/vim-autoformat, which supports lots of file types. I recently made a pull request for Typescript support using vvakame/typescript-formatter and it was accepted.

It still doesn't solve this annoyance and I'd rather avoid adding yet another plugin :]. I'll see if I can patch this and make a pull request.

**Comment by isaachess** [comment]
I'd rather we just remove indentation from this package altogether (as is stated in the readme).

**Comment by dani-h** [comment]
Aside from the syntax support file that Microsoft released standalone, what does this plugin really do? QuickFix and Compilation?

**Comment by dani-h** [comment]
@isaachess Not sure if this solves ur problem, but you can try. It doesn't remove the indentation though
b58c6a1c47ac852d7e2b67eaf4962a16d4a8b2ec

**Comment by leafgarland** [comment]
@isaachess not just weird, it's pretty poor, I apologise. I originally felt that this should not include indentation because it needs to be right and I don't have time for it but then accepted a PR that included custom indentation and did not update readme. I am travelling with very poor internet access so can only comment for now. Will take a look in a week or so.

**Comment by isaachess** [comment]
@leafgarland No problem, I know as well as anyone how easy it is to let a readme get out of date.

Can I suggest a setting to disable indentation with this package, in case we want to allow another package (such as vim-js-indent) to take care of the indentation.

**Comment by leafgarland** [comment]
Yeah, a disable flag is a good idea.

**Comment by leafgarland** [comment]
Fixed by #40
